stdlib/public/TensorFlow/CompilerRuntime.swift

```diff
@@ -767,14 +937,41 @@ public final class _TensorComputation {
 }
 
 private extension _TensorComputation {
-  /// Runs the tensor program. Aborts the process on error, and emits an error
-  /// string to STDERR.
+  /// In eager mode, runs a device function in a corresponding thread given by
+  /// `threadIdx`. Otherwise, runs all devices functions of the tensor
+  /// program. Aborts the process on error, and emits an error string to STDERR.
   // NOTE: This is to be called by the initializer. The computation gets
   // executed on initialization, thus this method will not be exposed to users.
-  private func execute() {
+  private func execute(threadIdx: Int) {
+    debugLog("Executing thread \(threadIdx).")
+    if let stateTFE = stateTFE {
+      internalConsistencyCheck(_RuntimeConfig.usesTFEagerAPI)
+      internalConsistencyCheck(threadIdx <= helperFunctionCount)
+      let op = stateTFE.ops[threadIdx]
+      if threadIdx == 0 {
+        var returnValueCount = Int32(returnValues.count)
+        TFE_Execute(op, &returnValues, &returnValueCount, status)
+        debugLog("""
+                   returnValues.count=\(returnValues.count), \
+                   returnValueCount=\(returnValueCount).
+                   """)
+        assert(returnValueCount == returnValues.count)
```

Comment:
Use `internalConsistencyCheck` for consistency with line 962.

Reply:
Good catch! Done.